docs/rules/no-ancestor-directory-import.md

```diff
@@ -0,0 +1,27 @@
+# Prefer that imports from within a directory extend to the file from where they are importing without relying on an index file. (no-ancestor-directory-import)
+
+Imports inside the same directory should extend directly to the file from where they are importing without relying on an index file. This means the source of these imports should not end with a directory (`/`), but the path should terminate at an individual filename. This preserves the index file inside a directory as a mechanism for exposing pieces of the module to the outside application.
```

Comment:
The last sentence should be augmented with `, rather than as a way to export the parts that the module internally depends on.` or something like that.